pkg/cmd/roachtest/test_runner.go

```diff
@@ -643,6 +648,14 @@ func (r *testRunner) runWorker(
 				}
 				qp.Release(alloc)
 				alloc = nil
+
+				// stop the tests if the failure rate has been exceeded
+				r.status.Lock()
+				failureRate := float64(len(r.status.fail)) / float64(r.testCount)
+				r.status.Unlock()
+				if failureRate > roachtestflags.AutoKillThreshold {
+					return errors.Errorf("failure rate %.2f exceeds limit %.2f", failureRate, roachtestflags.AutoKillThreshold)
+				}
```

Comment:
I'm confused -- why do we have this logic here? It is guarded by this predicate:

```go
// We could not find a test that can reuse the cluster. Destroy the cluster
// and search for a new test.
if testToRun.noWork {
```

Shouldn't we be aborting runs regardless of cluster reuse?

Reply:
I think it should suffice to check the threshold immediately after a test candidate was selected and before it's executed: https://github.com/cockroachdb/cockroach/blob/6210c2cd725ecb3875cad722020ea7c1d9042cec/pkg/cmd/roachtest/test_runner.go#L658-L661

Reply:
From my discussion with Renato I think moving it [right after the](https://github.com/cockroachdb/cockroach/blob/6210c2cd725ecb3875cad722020ea7c1d9042cec/pkg/cmd/roachtest/test_runner.go#L598) `select` block should also be fine

Reply:
Yep, that should work too since _all_ clusters are destroyed (`r.cr.destroyAllClusters(ctx, l)`) upon an exceptional termination of a worker.